### PR TITLE
stacer: init at 1.1.0

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
wanted a nice looking monitoring tool

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

```
$ nix path-info -Sh ./result
/nix/store/gagj7l3izf0vs91p42cp312d9lrzigv5-stacer-1.1.0	 283.7M
```

### CODE REVIEW

**Comment by dtzWill** [comment]
LGTM, ran it-- it does look nice!

Some of the features don't make sense for NixOS, so didn't try them to find out how that went but I don't think there's anything to be done about that.

Haven't looked through the expression yet, will do it in the morning if not merged by then already ;).

**Comment by jonringer** [comment]
@GrahamcOfBorg build stacer

File: pkgs/tools/admin/stacer/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = with lib; {
+    description = "Linux System Optimizer and Monitoring";
+    homepage = "https://github.com/oguzhaninan/Stacer";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ jonringer ];
+  };
+}
+
```
Comment (worldofpeace) [inline pkgs/tools/admin/stacer/default.nix]:
```suggestion
```

File: pkgs/tools/admin/stacer/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0qndzzkbq6abapvwq202kva8j619jdn9977sbqmmfs9zkjz4mbsd";
+  };
+
+  nativeBuildInputs = [ cmake qttools ];
+
+  buildInputs = [ qtcharts ];
+
+  enableParallelBuilding = true;
```
Comment (worldofpeace) [inline pkgs/tools/admin/stacer/default.nix]: this is done by default with cmake
```suggestion
```

**Comment by worldofpeace** [comment]
The program is not completely hermetic because it tries to launch executables out of band with  https://github.com/oguzhaninan/Stacer/search?p=1&q=CommandUtil%3A%3Aexec&unscoped_q=CommandUtil%3A%3Aexec

I'm not sure if it's ok for the administration tool to have hardcoded paths to executables where appropriate, because some could be optional features or ones, as @dtzWill, don't make sense in NixOS. Example of what I recommend https://github.com/NixOS/nixpkgs/issues/65252#issuecomment-538781768. It's also cool to inject into `PATH` via the wrapper, but that's less reliable.

**Comment by jonringer** [comment]
> The program is not completely hermetic because it tries to launch executables out of band with https://github.com/oguzhaninan/Stacer/search?p=1&q=CommandUtil%3A%3Aexec&unscoped_q=CommandUtil%3A%3Aexec

I'm fine with wrapping, it does look like they use any paths, just command names

**Comment by worldofpeace** [comment]
I'm having problems with it starting/using on my machine like https://github.com/oguzhaninan/Stacer/issues/349. Strace reveals weird stuff not related to any nixpkgs shenanigans so I'm going to assume it's an undiscovered bug/s upstream.

**Comment by jonringer** [comment]
I'm not sure the nature of the issues, closing as I've lost interest in this

**Pull request closed without merging**